src/modules/hero/hero.component.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { DebugElement } from '@angular/core';
+import { expect } from '@blackbaud/skyux-builder/runtime/testing/browser';
+
+import { StacheHeroComponent } from './hero.component';
+
+describe('StacheHeroComponent', () => {
+  let component: StacheHeroComponent;
+  let fixture: ComponentFixture<StacheHeroComponent>;
+  let debugElement: DebugElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [ StacheHeroComponent ]
+    })
+    .compileComponents();
+
+    fixture = TestBed.createComponent(StacheHeroComponent);
+    component = fixture.componentInstance;
+    debugElement = fixture.debugElement;
+  });
+
+  it('should take a whole number and convert it to a valid decmial', () => {
+    component.overlayOpacity = '5';
+    expect(component.overlayOpacity).toEqual('0.05');
+    component.overlayOpacity = '50';
+    expect(component.overlayOpacity).toEqual('0.5');
+    component.overlayOpacity = '1500';
+    expect(component.overlayOpacity).toEqual('1');
+    component.overlayOpacity = '-200';
+    expect(component.overlayOpacity).toEqual('0');
+  });
+
+  it('should accept a decimal and convert it to a valid 100th place decimal.', () => {
+    component.overlayOpacity = '0.5';
+    expect(component.overlayOpacity).toEqual('0.5');
+    component.overlayOpacity = '.03';
+    expect(component.overlayOpacity).toEqual('0.03');
+    component.overlayOpacity = '0.12456';
+    expect(component.overlayOpacity).toEqual('0.12');
+  });
+
+  it('should default to 0.4 given an invalid input', () => {
+    component.overlayOpacity = 'Sam Wise Gamgee';
```

Comment:
Because we're writing a public-facing library, it's a better practice to create "boring" tests, and avoid humor/cultural references. In other words, use something like `invalid value` instead of `Sam Wise Gamgee` :)

I think Hare and myself wrote a few of these in a similar fashion before; we'll need to go back and clean those up before the demo, as well. I'll create an issue for that.